### PR TITLE
Use stencil buffer for clipping iframes and aabb tree nodes.

### PR DESCRIPTION
**Pull request by glennw**
(This allows correct clipping for these cases when transforms are involved).

This is a partial fix for #177 (the other part relies on changes to layout).

### CODE REVIEW

**Comment by glennw** [comment]
r? @pcwalton 

(This relies on a related PR in servo before it can be merged).

**Comment by pcwalton** [comment]
Could we file a followup to add the number of stencil buffer changes to `debug_render`? It would be nice to be able to diagnose the problem quickly if we see pages in the wild that thrash the stencil buffer.

File: src/util.rs [hunk 1]
```diff
@@ -64,6 +64,19 @@ impl MatrixHelpers for Matrix4 {
...
+pub trait RectHelpers {
+    fn contains_rect(&self, other: &Rect<f32>) -> bool;
+}
+
+impl RectHelpers for Rect<f32> {
+    fn contains_rect(&self, other: &Rect<f32>) -> bool {
+        self.origin.x <= other.origin.x &&
+        self.origin.y <= other.origin.y &&
+        self.origin.x + self.size.width >= other.origin.x + other.size.width &&
+        self.origin.y + self.size.height >= other.origin.y + other.size.height
```
Comment (pcwalton) [inline src/util.rs]: nit: Could use `max_x()` and `max_y()` I think

**Comment by pcwalton** [comment]
Looks good.

**Comment by glennw** [comment]
Done - https://github.com/glennw/webrender/issues/181

**Pull request merged**